src/xpcc/architecture/platform/driver/core/avr/ram.cpp.in

```diff
@@ -76,9 +77,9 @@ initializeMemory(void)
 void *
 xpcc::avr::allocateMemory(std::size_t requestedSize)
 {
-	if (requestedSize == 0 ||
-		requestedSize > MAX_BLOCK_PARTS * BLOCK_SIZE)
+	if (requestedSize == 0 || requestedSize > MAX_BLOCK_PARTS * BLOCK_SIZE)
 	{
+		xpcc_assert(0, "core", "heap", "malloc", requestedSize);
```

Comment:
In c++ the new operator should throw a std::bad_alloc, since xpcc does not support exceptions, i think the assert is as close as it gets to c++ conformity. In order to have a c variant, it should return null, there is no nullptr in c